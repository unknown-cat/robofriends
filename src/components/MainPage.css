/* #### Generated By: http://www.cufonfonts.com #### */

@font-face {
  font-display: swap;
  font-family: "SEGA LOGO FONT";
  font-style: normal;
  font-weight: normal;
  src: local("SEGA LOGO FONT"), url("SEGA.woff") format("woff");
}

h1 {
  font-family: "SEGA LOGO FONT", sans-serif;
  font-weight: 200;
  color: #0ccac4;
}

@media (max-width: 420px) {
  .f1 {
    font-size: 2.5rem;
  }
}