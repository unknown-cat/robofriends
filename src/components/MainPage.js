import React, { Component } from 'react';

import CardList from './CardList';
import SearchBox from './SearchBox';
import Scroll from './Scroll';
import ErrorBoundry from './ErrorBoundry';
import Header from './Header';

import './MainPage.css';

class MainPage extends Component {
	constructor(props) {
		super(props)
		this.state = {
			count: 1
		}
	}
	componentDidMount() {
		this.props.onRequestRobots();
	}

	render() {
		const { robots, searchField, onSearchChange, isPending } = this.props;
		const filteredRobots = robots.filter(robot => {
			return robot.name.toLowerCase().includes(searchField.toLowerCase());
		})
		return (
			<div className='tc'>
				<Header/>
				<SearchBox searchChange={onSearchChange}/>
				<Scroll>
					{ isPending ? <h1>Loading</h1> :
						<ErrorBoundry>
							<CardList robots={filteredRobots} />
						</ErrorBoundry>
					}
				</Scroll>
			</div>
		);
	}
}

export default MainPage
