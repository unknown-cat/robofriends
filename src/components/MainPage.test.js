import React from "react";
import { shallow } from "enzyme";
import MainPage from "./MainPage";

let wrapper;
beforeEach(() => {
	const mockProps = {
		onRequestRobots: jest.fn(),
		robots: [],
		searchField: '',
		pending: false
	}

	wrapper = shallow(<MainPage { ...mockProps }/>)
})

it('renders mainPage', () => {
	expect(wrapper).toMatchSnapshot()
})

it('filters robots correctly', () => {
	const mockProps2 = {
		onRequestRobots: jest.fn(),
		robots: [{
			id: 3,
			name: 'john',
			email: 'john@gmail.com'
		}],
		searchField: 'john',
		pending: false
	}
	const wrapper2 = shallow(<MainPage { ...mockProps2 }/>)
	expect(wrapper2.instance().filterRobots()).toEqual([{
		id: 3,
		name: 'john',
		email: 'john@gmail.com'
	}])
})

it('expect to not found any robots after search', () => {
	const mockProps3 = {
		onRequestRobots: jest.fn(),
		robots: [{
			id: 3,
			name: 'john',
			email: 'john@gmail.com'
		}],
		searchField: 'a',
		pending: false
	}
	const filteredRobots = []
	const wrapper3 = shallow(<MainPage { ...mockProps3 }/>)
	expect(wrapper3.instance().filterRobots()).toEqual(filteredRobots)
})

it("expect to show the loading page if pending", () => {
	wrapper.setProps({ isPending: true })
	expect(wrapper).toMatchSnapshot("<h1>Loading</h1>")
})