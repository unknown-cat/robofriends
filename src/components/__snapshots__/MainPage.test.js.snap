// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to show the loading page if pending: <h1>Loading</h1> 1`] = `
<div
  className="tc"
>
  <Header />
  <SearchBox />
  <Scroll>
    <h1>
      Loading
    </h1>
  </Scroll>
</div>
`;

exports[`renders mainPage 1`] = `
<div
  className="tc"
>
  <Header />
  <SearchBox />
  <Scroll>
    <ErrorBoundary>
      <CardList
        robots={Array []}
      />
    </ErrorBoundary>
  </Scroll>
</div>
`;
